src/UI/Components/Diagram/Context/EventWrapper.tsx

```diff
@@ -0,0 +1,189 @@
+   * as the related instances aren't added to the serviceOrderItems map as we don't want to accidentally delete or edit them,
+   * yet they can exist on the canvas and be removed from it.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleUpdateServiceOrderItems = (event): void => {
+    const customEvent = event as CustomEvent;
+    const { cell, action } = customEvent.detail as {
+      cell: ServiceEntityBlock;
+      action: ActionEnum;
+    };
+
+    setServiceOrderItems((prev) => {
+      // related instances aren't added to the serviceOrderItems map, this condition is here to prevent situation where we try to remove the related instance from the canvas and it ends up here with status to delete it from the inventory
+      if (prev.has(String(cell.id)) || action === ActionEnum.CREATE) {
+        return updateServiceOrderItems(cell, action, prev);
+      }
+
+      return prev;
+    });
+  };
+
+  /**
+   * Handles updating the stencil state for the embedded entities.
+   * If the current count of the instances created from adequate stencil is more than or equal to the max count,, the adequate stencil element will become disabled.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleUpdateStencilState = (event): void => {
+    const customEvent = event as CustomEvent;
+    const eventData: { name: string; action: EmbeddedEventEnum } =
+      customEvent.detail;
+
+    //event listener doesn't get updated state outside setStencilState function, so all logic has to be done inside it
+    setStencilState((prev) => {
+      const stencilStateCopy = JSON.parse(JSON.stringify(prev));
+
+      const name = eventData.name;
+      const stencil = stencilStateCopy[name];
+
+      // If the stencil doesn't exist, return the previous state - that's the case when we add related instance through appendInstance() function which is related through it's own embedded entity - then the stencil state doesn't have that stencil stored
+      if (!stencil) {
+        return stencilStateCopy;
+      }
+
+      switch (eventData.action) {
+        case "add":
+          stencil.current += 1;
+          break;
+        case "remove":
+          stencil.current -= 1;
+          break;
+        default:
+          break;
+      }
+      const elements = [
+        { selector: `.${name}_body`, class: "stencil_accent-disabled" },
+        { selector: `.${name}_bodyTwo`, class: "stencil_body-disabled" },
+        { selector: `.${name}_text`, class: "stencil_text-disabled" },
+      ];
+
+      const shouldDisable =
+        stencil.max !== null &&
+        stencil.max !== undefined &&
+        stencil.current >= stencil.max;
+
+      // If the current count of the instances created from given stencil is more than or equal to the max count, disable the stencil of given embedded entity
```

Comment:
update based on feedback above.

Reply:
In case of what I wrote above, there is nothing to update, right?

Reply:
`// As in the docstrings mentioned, If the current count of the instances created from given stencil is more than or equal to the max count, disable the stencil of given embedded entity` 

that's what's in the code at the moment, but the view here isn't updated - I dunno why this keeps happening. is that enough?